app/views/posts/_post_info.html.erb

```diff
@@ -7,6 +7,10 @@
     Comments: <%= post.comments.size %>,
     Likes: <%= post.likes_counter.to_i %>
   </p>
+
+  <% if can? :destroy, @post %>
+    <%= button_to 'Delete Post', user_post_path(@post.author, @post), method: :delete, data: { confirm: 'Are you sure you want to delete this post?' }, class: 'delete-button' %>
+  <% end %>
```

Comment:
- [ ] When I try to create a new `Post` instance, I get this error:

![image](https://github.com/mershark/blog-app/assets/77106962/6323afe1-9adc-4f76-890f-07fb15e2242a)

When using the route path for the delete button, you are using `@post` instead of `post`. Remember that in the partial that you created, you are passing `post` as a local variable and not `@post`. Kindly change it. 👍 😎